tensorflow_lite_support/python/test/task/text/bert_clu_annotator_test.py

```diff
@@ -37,33 +37,62 @@
 _BertCluAnnotationOptions = clu_annotation_options_pb2.BertCluAnnotationOptions
 
 _BERT_MODEL = 'bert_clu_annotator_with_metadata.tflite'
+
 _CLU_REQUEST = _CluRequest(utterances=[
-    'I would like to make a restaurant reservation at morning 11:15.',
-    'Which restaurant do you want to go to?',
-    'Can I get a reservation for two people at Andes Cafe? Where is their ' +
-    'address?'
+  'Hi, I would like to make a dinner reservation.'
```

Comment:
You're using one long sentence. Please split them into sentences where speakers change. Without it, the max_history_turns won't have any effect because it see everything as one turn.

For example:
```
'Hi, I would like to make a dinner reservation.',
'Of course, what evening will you be joining us on?'
```

Reply:
@khanhlvg Oh sorry about that. I've fixed it now. PTAL again.

Reply:
Thanks Kinar for the quick fix.

As a sanity check, could you test directly using the C++ API and see whether the same list of utterances with the default config gives the same result with the C++ API? From the expected response, I see that the model wasn't able to pick up info such as reservation dates, room type etc., which are important for a hotel reservation. It can well be the limitation of the model itself, but it's safer to check and make sure that it's not a bug in the Python implementation.

Reply:
@khanhlvg The C++ API seems to generate the same results from my testing. So I believe it has to do with the model itself not being able to catch all of the info. You can feel free to try it out yourself too. I just happened to replace a [test case](https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/cc/test/task/text/bert_clu_annotator_test.cc#L129-L157) with the following lines of code and it should give the same results as you can see in the Python API:

```cpp
TEST(BertCluAnnotatorTest, TestAnnotatorBasic) {
  BertCluAnnotatorOptions options;
  options.mutable_base_options()->mutable_model_file()->set_file_name(
      GetFullPath(kTestBertCluAnnotatorModelWithMetadataPath));
  SUPPORT_ASSERT_OK_AND_ASSIGN(std::unique_ptr<CluAnnotator> clu_annotator,
                       BertCluAnnotator::CreateFromOptions(options));
  CluRequest request;
  ASSERT_TRUE(TextFormat::ParseFromString(
      R"pb(
            utterances: "I would like to book a reservation at your hotel"
            utterances: "What date would you like to make that reservation for?"
            utterances: "I need the reservation for the 14th of May"
            utterances: "How many days do you need the reservation for?"
            utterances: "I will be staying for 3 nights."
            utterances: "Is that a single room, or will there be more guests?"
            utterances: "I need a double room."
            utterances: "We have smoking and nonsmoking rooms. Which do you prefer?"
            utterances: "We require a nonsmoking room."
            utterances: "Your room is booked. You must arrive before 4:00 pm the day you are to check in."
      )pb",
      &request));
  SUPPORT_ASSERT_OK_AND_ASSIGN(const auto response, clu_annotator->Annotate(request));

  ExpectApproximatelyEqual(
      response, ParseTextProtoOrDie<CluResponse>(R"pb(
            domains {
              index: 0
              score: 0.915805
              display_name: "Hotels"
              class_name: ""
            }
            intents {
              index: 0
              score: 0.611646
              display_name: "NOTIFY_SUCCESS"
              class_name: ""
            }
            noncategorical_slots {
              slot: "time"
              extraction {
                value: "4:00 pm"
                score: 0.794008
                start: 44
                end: 51
              }
            }
      )pb"));
}
```

Reply:
Thanks Kinar for verifying. We're still waiting for internal review so I'll keep you updated on how it goes.